src/modules/Game.class.js

```diff
@@ -1,68 +1,236 @@
+
+      for (let j = 0; j < size; j++) {
+        if (isVertical) {
+          state[j][i] = newLine[j];
+        } else {
+          state[i][j] = newLine[j];
+        }
+      }
+    }
+
+    return this.isMove(oldState, state);
+  }
+
+  isMove(oldState, newState) {
+    for (let i = 0; i < oldState.length; i++) {
+      for (let j = 0; j < oldState.length; j++) {
+        if (oldState[i][j] !== newState[i][j]) {
+          return true; // if there is a diff in state
+        }
+      }
+    }
+
+    return false; // if there is no diff in state
+  }
+
+  mergeLine(line, direction, { precededCheckMove }) {
+    const size = line.length;
+    const stack = [];
+    const newLine = new Array(size).fill(0);
+
+    const startIndex =
+      direction === 'up' || direction === 'left' ? 0 : size - 1;
+    const endIndex = direction === 'up' || direction === 'left' ? size : -1;
+    const step = direction === 'up' || direction === 'left' ? 1 : -1;
+
+    let newIndex = startIndex;
+
+    for (let i = startIndex; i !== endIndex; i += step) {
+      const cellValue = line[i];
+
+      if (cellValue !== 0) {
+        if (stack.length === 0) {
+          stack.push(cellValue);
+        } else {
+          if (stack[0] === cellValue) {
+            const mergedValue = +stack[0] + +cellValue;
+
+            newLine[newIndex] = stack.pop() + cellValue;
+            newIndex += step;
+
+            if (!precededCheckMove) {
+              this.score += mergedValue;
+            }
+          } else {
+            newLine[newIndex] = stack.pop();
+            newIndex += step;
+            stack.push(cellValue);
+          }
+        }
+      }
+    }
+
+    if (stack.length > 0) {
+      newLine[newIndex] = stack.pop();
+    }
+
+    return newLine;
+  }
+
+  getScore() {
+    return this.score;
+  }
+
+  getState() {
+    return this.state;
+  }
+
+  getStateCopy() {
+    return this.state.map((row) => [...row]);
+  }
+
+  getStatus() {
+    if (this.checkWin()) {
+      this.status = 'win';
+    } else if (this.status === 'playing' && !this.hasMoreMove()) {
+      this.status = 'lose';
+    }
+
+    return this.status;
+  }
+
+  checkWin() {
+    return this.state.some((line) => line.includes(2048));
+  }
+
+  start() {
+    this.status = 'playing';
+    this.generateNewCell();
+    this.generateNewCell();
+  }
+
+  restart() {
+    this.state = Game.getInitialState();
+    this.status = 'idle';
+    this.score = 0;
+  }
+
+  updateField(field) {
+    for (let i = 0; i < field.rows.length; i++) {
+      for (let j = 0; j < field.rows[i].cells.length; j++) {
+        const cellValue = this.state[i][j];
+
+        field.rows[i].cells[j].textContent = cellValue !== 0 ? cellValue : '';
+
+        if (cellValue !== '') {
+          field.rows[i].cells[j].setAttribute(
+            'class',
+            `field-cell field-cell--${cellValue}`,
+          );
+        }
+      }
+    }
+
+    const fragment = document.createDocumentFragment();
```

Comment:
there is no `document` in tests (they run un node.js, not in browser), so tests for Game.class.js might fail because of that
If you need help feel free to ask in the chat